reconcile/utils/gql.py

```diff
@@ -218,7 +232,30 @@ def commit_timestamp_utc(self) -> Optional[str]:
         return None
 
 
-_gqlapi: Optional[GqlApi] = None
+class GqlApiSingleton:
+    gql_api: Optional[GqlApi] = None
+    gqlapi_lock = threading.Lock()
+
+    def __new__(cls, *args, **kwargs):
+        if cls.gql_api:
+            logging.debug("Resestting GqlApi instance")
+            cls.close()
+        with cls.gqlapi_lock:
+            cls.gql_api = GqlApi(*args, **kwargs)
```

Comment:
There is an edge case that another thread can acquire the lock and create a new `GqlApi`, it will be override without close, should guard the whole process with lock:

```python
    def __new__(cls, *args, **kwargs):
        with cls.gqlapi_lock:
            if cls.gql_api:
                logging.debug("Resestting GqlApi instance")
                cls.gql_api.close()
            cls.gql_api = GqlApi(*args, **kwargs)
            return cls.gql_api # <- return instance object
```